src/main/java/de/tum/in/www1/artemis/service/TextBlockService.java

```diff
@@ -49,23 +46,24 @@
      * @param submission TextSubmission to split
      * @return List of TextBlocks
      */
-    @Transactional(readOnly = true)
-    public List<TextBlock> splitSubmissionIntoBlocks(TextSubmission submission) {
+    public Set<TextBlock> splitSubmissionIntoBlocks(TextSubmission submission) {
+        final Set<TextBlock> blocks = new HashSet<>();
+
+        // Return empty set for missing submission text.
         final String submissionText = submission.getText();
-        if (submissionText == null)
-            return new ArrayList<>();
-        // Return empty list for missing submission text.
+        if (submissionText == null) {
+            return blocks;
```

Comment:
Would it make sense to use [Collections.emptySet()](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#emptySet%28%29) here, which avoids allocating a new `HashSet` if we don't need it?

Reply:
changed it, but I don't think it will make a noticeable difference (maybe 1ns faster ;-)